src/AppBundle/Form/Ad/AddUserType.php

```diff
@@ -8,54 +8,30 @@
 
 class AddUserType extends AbstractType
 {
-    /**
-     * @var array
-     */
-    private $options;
-
-    /**
-     * @param array $options keys: array roleChoices, array roleNameEmptyValue
-     */
-    public function __construct(array $options)
-    {
-        $this->options = $options;
-    }
-
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $roleNameOptions = [
-            'choices' => $this->options['roleChoices'],
-            'empty_value' => $this->options['roleNameEmptyValue'],
-        ];
-
-        if (!empty($this->options['roleNameSetTo'])) {
-            $roleNameOptions['data'] = $this->options['roleNameSetTo'];
-        }
-
-        if (!empty($this->options['roleNameDisabled'])) {
-            $roleNameOptions['disabled'] = 'disabled';
-        }
-
-        $odrEnabledOptions =[];
-        if (!empty($this->options['odrEnabledDisabled'])) {
-            $odrEnabledOptions['disabled'] = 'disabled';
-        }
-        $odrEnabledOptions['data']=true;
-
         $builder
             ->add('firstname', 'text')
             ->add('lastname', 'text')
-            ->add('roleName', 'choice', $roleNameOptions)
-            ->add('odrEnabled', 'checkbox', $odrEnabledOptions)
+            ->add('roleName', 'choice', [
+                    'empty_value' => null,
+                    'choices'     => $options['options']['roleChoices'],
+                    'data'        => $options['options']['roleNameSetTo'],
```

Comment:
If you're passing a single array of options these could be $options['roleChoices'], $options['roleNameSetTo'] etc

Reply:
yes, exactly what I'm planning to simplify after I write a behat test (that will guarantee I don't mess it up)